api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.oauth2.v3;
+
+import "envoy/config/route/v3/route_components.proto";
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.oauth2.v3";
+option java_outer_classname = "OauthProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: OAuth]
+// OAuth :ref:`configuration overview <config_http_filters_oauth>`.
+// [#extension: envoy.filters.http.oauth2]
+//
+
+message OAuth2Credentials {
+  // The client_id to be used in the authorize calls. This value will be URL encoded when sent to the OAuth server.
+  string client_id = 1 [(validate.rules).string = {min_bytes: 1}];
+
+  // The secret used to retrieve the access token. This value will be URL encoded when sent to the OAuth server.
+  transport_sockets.tls.v3.SdsSecretConfig token_secret = 2
+      [(validate.rules).message = {required: true}];
+
+  // Configures how the secret token should be created.
+  oneof token_formation {
+    option (validate.required) = true;
+
+    // If present, the secret token will be a HMAC using the provided secret.
+    transport_sockets.tls.v3.SdsSecretConfig hmac_secret = 3;
+  }
+}
+
+// OAuth config
+//
+// [#next-free-field: 10]
+message OAuth2Config {
+  // Name of the cluster to send OAuth requests to.
+  string cluster = 1 [(validate.rules).string = {min_bytes: 1}];
```

Comment:
The combination of `cluster`/`hostname`/`token_path`/`timeout` is re-inventing `envoy.config.core.v3.HttpUri`, this should be `envoy.config.core.v3.HttpUri token_endpoint`. Naming it token_endpoint to match the OAuth 2 spec RFC.

Reply:
So seems like HttpUri wraps the path + hostname into a single uri field, but this filter will sometimes use just the hostname (to construct the redirection URL). Seems like we can either require configure the redirect and token URLs independently, or parse out the hostname from the token_endpoint field and use that to construct the redirect URL. WDYT?

Reply:
I'll go with `HttpUri token_endpoint` + `string redirection_hostname` for now, but happy to change